### DISCUSSION TITLE
Two-factor authentication required for Firefox extension developer accounts

### DISCUSSION THREAD

**Discussion by webguyio** [Authenticator-Extension/Authenticator#619] [General]
Mozilla just put out the call that 2FA is now required.

I'm not able to get this to work for either scanning or manual entry on https://accounts.firefox.com/settings/two_step_authentication.

I know that add-ons are disabled to a certain extent on mozilla.org, so perhaps this is simply not possible, although from the screenshots, it looks like it was at some point?

https://addons.cdn.mozilla.net/user-media/previews/full/208/208795.png

**Comment by mymindstorm** [comment]
Yes, extensions are blocked from the add-ons and Firefox website (i.e. qr scan and autofill won't work), but I believe there is a setting in `about:config` you can use to bypass this. In any case you can always use manual entry.

Could you give more details on how you are having problems with manual entry?

**Comment by webguyio** [comment]
After entering the secret, it begins generating verification codes, but when entered...

https://calmestghost.com/screen.jpg

(actual data redacted)

I wasn't able to find the preference name, but I'd be happy to try the bypass if you can recall what it was?

**Comment by mymindstorm** [comment]
This seems to be a problem with your computer time, I was able to change my 2FA secret successfully using manual entry. Could you try using the "Sync Time with Google" menu option and trying again?

**Comment by webguyio** [comment]
That worked, but now my Mozilla account won't save the enabled state for 2FA in my settings. Switching to the new firefox.com design let me save, but was again disabled as soon as I tried logging out and logging back in.

Please note that this issue is isolated to Mozilla/Firefox only, and the add-on has worked fine with all other sites I've tried, without these issues.

I'll fuss with this some more tomorrow.

Thank you for your help.

**Comment by mymindstorm** [comment]
It may be helpful to take a video of the issue and file a bug against Firefox accounts [here](https://bugzilla.mozilla.org/enter_bug.cgi?format=guided#h=dupes%7CCloud+Services%7CServer%3A+Firefox+Accounts).
